### PR TITLE
Frontend Test-Eeron: Sprint 3 Pull 1

### PR DESCRIPTION
**Pull request by ItisEeron**
A pull requests to the main test branch. This branch will be in use after this pull. I've added many tests to the redux store. I've attempted to organize them in a meaningful manner. I can now focus on testing components and pages.

### CODE REVIEW

File: homepairs/homepairsUI/src/Screens/Components/Modals/AddNewPropertyModal/AddNewPropertyModalBase.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import strings from 'homepairs-strings';
 import * as BaseStyles from 'homepairs-base-styles';
 import { HomePairsDimensions, Property, AddNewPropertyState } from 'homepairs-types';
 import Colors from 'homepairs-colors';
-import {isNumber} from 'homepairs-utilities';
+import {isPositiveWholeNumber} from 'homepairs-utilities';
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/Screens/Components/Modals/AddNewPropertyModal/AddNewPropertyModalBase.tsx]: better name

File: homepairs/homepairsUI/src/Screens/Components/Modals/EditPropertyModal/EditPropertyModalBase.tsx [hunk 2]
```diff
@@ -210,7 +210,7 @@ export default class EditNewPropertyModalBase extends React.Component<Props, Edi
     validateForms() {
         const {address, city, state, bedrooms, bathrooms, tenants} = this.state;
         if (!isEmptyOrSpaces(address) && !isEmptyOrSpaces(city) && !isEmptyOrSpaces(state)) {
-            if (isNumber(bedrooms) && isNumber(bathrooms) && isNumber(tenants)) {
+            if (isPositiveWholeNumber(bedrooms) && isPositiveWholeNumber(bathrooms) && isPositiveWholeNumber(tenants)) {
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/Screens/Components/Modals/EditPropertyModal/EditPropertyModalBase.tsx]: A property could, perhaps, technically not have bedrooms

File: homepairs/homepairsUI/src/state/account/reducer.tsx [hunk 3]
```diff
@@ -1,25 +1,37 @@
...
+ * root store should ever need to use this function.
+ * @param {AccountState} state - current state of the redux-store
+ * @param {AccountStateAction} action - the value that determines what type of mutation will occur with the store 
+ */
 export const accountProfile = (
     state: AccountState = initialState,
     action: AccountStateAction,
 ) => {
     let profileState: AccountState;
-    const prevState = {...state, modalOpen: true };
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/state/account/reducer.tsx]: Can't verify if this is a good change or not. Might want to ask Cesar

File: homepairs/homepairsUI/src/state/property-list/actions.tsx [hunk 4]
```diff
@@ -46,26 +81,19 @@ export const postNewProperty = (
...
+              token: info.roopairsToken,
+            })
             .then(response => {
-                if (
-                    !(
-                        response[responseKeys.DATA][responseKeys.STATUS] ===
-                        responseKeys.STATUS_RESULTS.FAILURE
-                    )
-                ) {
+                if (response[responseKeys.DATA][responseKeys.STATUS] === responseKeys.STATUS_RESULTS.SUCCESS) {
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/state/property-list/actions.tsx]: much better

File: homepairs/homepairsUI/src/state/header/reducer.tsx [hunk 5]
```diff
@@ -42,7 +43,7 @@ export const header = (
     action: HeaderAction, 
 ) => {
     /* * NOTE: USE IMMUTABLE UPDATE FUNCTIONS FOR REDUCERS OR ELSE REDUX WILL NOT UPDATE!!! * */
-    const newState = {...state, modalOpen: true };
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/src/state/header/reducer.tsx]: same as before, consult Cesar if this is a good change

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx [hunk 6]
```diff
@@ -206,51 +211,58 @@ describe('FetchAccount Action', () => {
...
             status: 'success',
             ...testJsonValues.TENANT_WITH_TOKEN,
             role: 'tenant',
         };
         mock.onPost('https://homepairs-alpha.herokuapp.com/API/login/').reply(200, data);
-        propertyManagerMock1.dispatch(AccountActions.fetchAccount(email, password, testProps.navigation)).then(() => {
+        await propertyManagerMock1.dispatch(AccountActions.fetchAccount(email, password, testProps.navigation)).then(() => {
           expect(spyFunction.call).toHaveLength(1);
           const actionResults = propertyManagerMock1.getActions();
           expect(actionResults).toHaveLength(2);
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: Can / should we check actionResults[0] as well (not just [1])?

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx [hunk 7]
```diff
@@ -260,10 +272,11 @@ describe('FetchAccount Action', () => {
         mock.onPost('https://homepairs-alpha.herokuapp.com/API/login/').reply(200, data);
         await propertyManagerMock1.dispatch(AccountActions.fetchAccount(email, password, testProps.navigation, statusFailedSpy)).then(() => {
           expect(propertyManagerMock1.getActions()).toHaveLength(0);
-          expect(statusFailedSpy.mock.calls).toHaveLength(1)
+          expect(statusFailedSpy.mock.calls).toHaveLength(1);
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: Should we check the specific failure at all? (not just the length)

File: homepairs/homepairsUI/src/state/header/reducer.tsx (hunk 5 above)
Comment (ItisEeron) [reply homepairs/homepairsUI/src/state/header/reducer.tsx]: This is fine. Remember the reducer state does not affect any of the modals used. We have given components the power of taking care of that. If one needs to close or open, our actions will invoke a components callback to deal with this.

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx (hunk 7 above)
Comment (ItisEeron) [reply homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: We could but we haven't generated all of our error messages yet. This is something Cesar can test. It would literally be a waste of time for me to do this at this point.

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx (hunk 6 above)
Comment (ItisEeron) [reply homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: We don't actually care about the actionResults since these had. We had tested these in previous suites. We simply care that the amount of dispatch functions on a successful response has been called. No need to make redundant test (which I already have plenty of)

File: homepairs/homepairsUI/src/state/account/reducer.tsx (hunk 3 above)
Comment (ItisEeron) [reply homepairs/homepairsUI/src/state/account/reducer.tsx]: modalOpen does nothing to our components. It should not be included in the state. Our components individually take care of their own modals. Any time a modal is affected outside of the component, it is via callback. Never the state.

Comment (YellowRainBoots) [reply homepairs/homepairsUI/src/state/account/reducer.tsx]: our modals are decoupled from our reducer, so their state shouldn't be tracked in the reducer anymore

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx (hunk 6 above)
Comment (YellowRainBoots) [reply homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: " tested these in previous suites" g2k

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx (hunk 7 above)
Comment (YellowRainBoots) [reply homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: sounds good, I agree

File: homepairs/homepairsUI/tests/utilities/SyntaxValidator.test.tsx [hunk 8]
```diff
@@ -159,4 +160,36 @@ describe('Test isEmptyOrSpaces Function', () =>{
         expect(isEmptyOrSpaces(value)).toBeFalsy();
     });
 
+});
+
+
+
+/** isPositiveWholeNumber Test * */
+const positiveWholeNumbers: string[] = [
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/utilities/SyntaxValidator.test.tsx]: Refactor - omit positive: positive whole numbers is a bad name. just stick with whole numbers (perhaps refactor) since whole numbers are already positive and zero. And you're saying 0 is a positiveWholeNumber, but zero is neither positive nor negative.

File: homepairs/homepairsUI/tests/utilities/SyntaxValidator.test.tsx [hunk 9]
```diff
@@ -159,4 +160,36 @@ describe('Test isEmptyOrSpaces Function', () =>{
...
+    '1',
+    '12345678910',
+    '90',
+    '10000',
+    '00000000',
+];
+
+const nonPositiveWholeNumbers: string[] = [
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/utilities/SyntaxValidator.test.tsx]: it'd be nice to test an edge case closer than -1.0 for instance, see where -0.99 or -0.5 or -0.49 or -.001 go

File: homepairs/homepairsUI/tests/utilities/SyntaxValidator.test.tsx [hunk 10]
```diff
@@ -159,4 +160,36 @@ describe('Test isEmptyOrSpaces Function', () =>{
...
+    '-10.3',
+];
+
+describe('Test isPositiveWholeNumber Function', () =>{
+    test.each(positiveWholeNumbers)('This value should return true', (value) => {
+        // All results are expected to be true for the first values
+        expect(isPositiveWholeNumber(value)).toBeTruthy();
+    });
+    test.each(nonPositiveWholeNumbers)('This value should return false', (value) => {
+        // All results are expected to be true for the first values
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/utilities/SyntaxValidator.test.tsx]: update this comment

File: homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+const testPropertyListStateWithIndex: PropertyListState = {
+    selectedPropertyIndex: 1,
+    properties: PMProperties,
+};
+
+describe('PropertyList Reducer Test', () => {
+
+    describe('Test action with Invalid Property type', () => {
+        it('Undefined State', () => {
+            const testAction: FetchPropertiesAction = {
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx]: this const testAction is literally repeated for both "it"s. just define it at the top once

File: homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+                type: 'INVALID_STATE',
+                properties: [],
+            };
+            const updatedProperties = properties(testPropertyListStateWithoutIndex, testAction);
+            expect(updatedProperties).toStrictEqual(testPropertyListStateWithoutIndex);
+        });
+    });
+
+    describe('Test action with ADD_PROPERTY type', () => {
+        const testProperty: Property = {
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx]: this already exists in PMProperties. Don't recreate variables you already have

File: homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+            bathrooms: 2,
+            bedrooms: 3,
+            state: 'CA',
+            streetAddress: '200 N. Santa Rosa',
+        };
+        const testAction: AddPropertyAction = {
+            type: PROPERTY_LIST_ACTION_TYPES.ADD_PROPERTY,
+            userData: testProperty,
+        };
+        it('Default State', () => {
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx]: these tests could be a tad more context (or im just bad with reducers) with what the default state is, since it'd offer explanation why remove/update don't have a default state. this certainly isn't required though

File: homepairs/homepairsUI/tests/redux/property-list/UpdatePropertyTest.test.tsx [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+const updatedPropertyCredentials: EditPropertyState = {
+    email: 'jackylynne@gmail.com',
+    index: 0,
+    oldProp: prevProperty,
+    roopairsToken: '1B88484KDKSLIEK058UDDD',
+};
+
+// To test nested actions with an async REST request, we need to call an await on the
+// dispatch methods in order to get an updated state.
+describe('postUpdatedProperty Action', () => {
```
Comment (YellowRainBoots) [inline homepairs/homepairsUI/tests/redux/property-list/UpdatePropertyTest.test.tsx]: I would love to see more success tests that show some fields changing  vs all fields changing vs no fields changing

Comment (YellowRainBoots) [reply homepairs/homepairsUI/tests/redux/property-list/UpdatePropertyTest.test.tsx]: We ended up discussing this and determined that separate allocation testing will occur for components and perhaps separate allocation methods for http requests

File: homepairs/homepairsUI/tests/redux/account/AccountReducer.test.tsx [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+      streetAddress: undefined,
+      city: undefined,
+      lastName: 'profile[accountKeys.LASTNAME]',
+      email: 'jacklame@gmail.com',
+      phone: '555555555',
+      roopairsToken: '',
+      manId: 102449555,
+    },
+  },
+  ACTION_WITH_NO_TYPE: { // PM Account with token
```
[the remark this answers is not in the document]
Comment (ItisEeron) [reply homepairs/homepairsUI/tests/redux/account/AccountReducer.test.tsx]: I'm not going to Change ACTION_WITH_NO_ TYPE. This it is literally a testament to the name of the action passed. I did rename, the PM to FETCH_PM and TENANT to FETCH_TENANT.

File: homepairs/homepairsUI/src/state/account/reducer.tsx (hunk 3 above)
Comment (cfchacon10) [reply homepairs/homepairsUI/src/state/account/reducer.tsx]: yeah modalOpen does not affect my modals. Should be fine.

File: homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx (hunk 7 above)
Comment (cfchacon10) [reply homepairs/homepairsUI/tests/redux/account/FetchAccount.test.tsx]: hmmm I don't handle these error messages tho. Not with what I've made at this point. I don't handle errors from the API calls.

File: homepairs/homepairsUI/tests/redux/account/GeneratePMAccount.test.tsx [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+            AccountActions.generateAccountForPM(testPMAccount1, password, testProps.navigation, statusFailedSpy))
+            .then(() => {
+                expect(statusFailedSpy.call).toHaveLength(1);
+                expect(propertyManagerMock1.getActions()).toHaveLength(0);
+                expect(statusFailedSpy).toBeCalledWith("Connection to the server could not be established.");
+            });
+      });
+
+      
+      it('When homepairs response returns failure', async () => {
```
[the remark this answers is not in the document]
Comment (cfchacon10) [reply homepairs/homepairsUI/tests/redux/account/GeneratePMAccount.test.tsx]: I guess I will get started on backend errors then lol
